### PR TITLE
Maven cache issue

### PR DESCRIPTION
**Pull request by tkvangorder**

### CODE REVIEW

**Comment by tkvangorder** [comment]
This is not perfect, but this changes the downloader to iterate over each repo until one of them resolves the raw maven file.

**Comment by tkvangorder** [comment]
This fixes #166

File: rewrite-maven/src/main/java/org/openrewrite/maven/cache/InMemoryCache.java [hunk 1]
```diff
@@ -53,7 +52,7 @@ private void fillUnresolvablePoms() {
         new BufferedReader(new InputStreamReader(MavenDownloader.class.getResourceAsStream("/unresolvable.txt"), StandardCharsets.UTF_8))
                 .lines()
                 .filter(line -> !line.isEmpty())
-                .forEach(gav -> pomCache.put(gav, Optional.empty()));
+                .forEach(unresolvablePoms::add);
```
Comment (jkschneider) [inline rewrite-maven/src/main/java/org/openrewrite/maven/cache/InMemoryCache.java:55]: I originally intended for `unresolvablePoms` to be an immutable denylist. Basically, the bad artifacts that have never worked that regularly get requested in maven central.

Comment (jkschneider) [reply rewrite-maven/src/main/java/org/openrewrite/maven/cache/InMemoryCache.java:55]: Seems like we should instead be keying pom cache on gav+repo and adding `Optional.empty()`.

**Review by sambsnyd — APPROVED** [review]
Great work, Tyler. We should replicate this fix across the other `MavenCache` implementations... lest we re-discover and re-investigate and re-fix this issue at some point later

File: rewrite-maven/src/main/java/org/openrewrite/maven/cache/InMemoryCache.java (hunk 1 above)
Comment (tkvangorder) [reply rewrite-maven/src/main/java/org/openrewrite/maven/cache/InMemoryCache.java:55]: I am a little confused about what you want to change here. 

I am collecting the unresolved group/artifact/versions into a string list now.

See below:

File: rewrite-maven/src/main/java/org/openrewrite/maven/cache/InMemoryCache.java [hunk 2]
```diff
@@ -81,8 +80,15 @@ private void fillUnresolvablePoms() {
...
-        String cacheKey = groupId + ':' + artifactId + ':' + version;
+
+        //There are a few exceptional artifacts that will never be resolved by the repositories. This will always
+        //result in an Unavailable response from the cache.
+        String artifactCoordinates = groupId + ':' + artifactId + ':' + version;
+        if (unresolvablePoms.contains(artifactCoordinates)) {
+            return UNAVAILABLE_POM;
+        }
+
+        String cacheKey = repo.toString() + ":" + artifactCoordinates;
```
Comment (tkvangorder) [inline rewrite-maven/src/main/java/org/openrewrite/maven/cache/InMemoryCache.java:91]: Note, I first create just the group/artifact/version key and check it against the unresolvable list. I return UNAVAILABEL_POM immediately if we encounter one of these poms. 

Then I add the repository to the key and check the cache. The cache will have an Optional.empty() if it is not found in the repository (and that also results in an UNAVAILABLE_POM.

**Pull request merged**